Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -639,13 +639,13 @@ ExceptionOr<Ref<NodeList>> LocalDOMWindow::collectMatchingElementsInFlatTree(Nod
 
     SelectorQuery& query = queryOrException.releaseReturnValue();
 
-    Vector<Ref<Element>> result;
-    for (auto& node : composedTreeDescendants(*scopeContainer)) {
-        if (RefPtr element = dynamicDowncast<Element>(node); element && query.matches(*element) && !node.isInUserAgentShadowTree())
-            result.append(element.releaseNonNull());
+    Vector<Ref<Node>> result;
+    for (Ref node : composedTreeDescendants(*scopeContainer)) {
+        if (auto* element = dynamicDowncast<Element>(node.get()); element && query.matches(*element) && !node->isInUserAgentShadowTree())
```

Comment:
Can we use the following here?
```
if (RefPtr element = dynamicDowncast<Element>(WTFMove(node)); element && query.matches(*element) && !element->isInUserAgentShadowTree())
    result.append(WTFMove(element));
```

And then keep Element type in the Vector? I don't understand why we're switching to Node type here.

Reply:
Sorry for the delay. It turned out the warning can be fixed in a simpler way.